From patchwork Tue May 18 19:21:31 2021
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Max Gurtovoy <mgurtovoy@nvidia.com>
X-Patchwork-Id: 12265521
Return-Path: <kvm-owner@kernel.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-19.1 required=3.0 tests=BAYES_00,DKIMWL_WL_HIGH,
	DKIM_SIGNED,DKIM_VALID,DKIM_VALID_AU,HEADER_FROM_DIFFERENT_DOMAINS,
	INCLUDES_CR_TRAILER,INCLUDES_PATCH,MAILING_LIST_MULTI,SPF_HELO_NONE,SPF_PASS,
	USER_AGENT_GIT autolearn=ham autolearn_force=no version=3.4.0
Received: from mail.kernel.org (mail.kernel.org [198.145.29.99])
	by smtp.lore.kernel.org (Postfix) with ESMTP id 1508FC433ED
	for <kvm@archiver.kernel.org>; Tue, 18 May 2021 19:21:40 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [23.128.96.18])
	by mail.kernel.org (Postfix) with ESMTP id E1DD560724
	for <kvm@archiver.kernel.org>; Tue, 18 May 2021 19:21:39 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1346055AbhERTW5 (ORCPT <rfc822;kvm@archiver.kernel.org>);
        Tue, 18 May 2021 15:22:57 -0400
Received: from mail-mw2nam12on2089.outbound.protection.outlook.com
 ([40.107.244.89]:64256
        "EHLO NAM12-MW2-obe.outbound.protection.outlook.com"
        rhost-flags-OK-OK-OK-FAIL) by vger.kernel.org with ESMTP
        id S237739AbhERTW5 (ORCPT <rfc822;kvm@vger.kernel.org>);
        Tue, 18 May 2021 15:22:57 -0400
ARC-Seal: i=1; a=rsa-sha256; s=arcselector9901; d=microsoft.com; cv=none;
 b=PCp0+QtBpnHOoMRXx3j9LIAD5sfJJf/BafVvsgcWTDYUYBnqwYO7G7kQ+CRSv8wGWDQpYdgSgRn4O0JdBU2DPDVAu397r78glc7drgp8n7SmNKgUwsQZuJ/fAXG8qGqSfzQU6mW2/lc3p1Q9CiN4EAW0mNhfY6aFawmEm61TAwKf0MXjQSRa0CMAivIGop37hzvgfFtxtQGqMHdMj4J+QO4FuDl6j0DRtSI2plwUaFDz9aVAXYCAAWBEnbhl9R4weUTh0wGKt4yKGGKYyqekBP6HNi2gw11RTLJiIw1Rbk88xipR7s0uJT2mi6/cjIaoHjI352T92GuxGwNBnl2rPA==
ARC-Message-Signature: i=1; a=rsa-sha256; c=relaxed/relaxed; d=microsoft.com;
 s=arcselector9901;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=KcnTnGk4MKucZ5I2jfckFtN2j5CTu6Fp125uCcm022A=;
 b=RMGkE8DCHHEr/hTkwtfskGjE9NNS/Fm2fhfXD0UAVFCilkXt5P2yDIJsHEdF3FSYA97OWudJ/XncVKK7tfhzalNTZvM2N+Ru6eQEJKSanJUEW/m6d8GR4Kz8RMin+dV9jnFEb7vAx8ezEOhoeW72CgT8ooc2d+RYPx0p7Pbcb7qkRTeb+2K+C/9IEq/56pFYA73FEnh6CM+OwkOiCcE9XSGJHbYqekKdbcLbCcBO35P78GXXmKPgplpZFGbMZ8bz5VmzHQCEUsTdwEY/edOvDM7VuVasGvaDflmY+KV7Tc2WAftEE0BnDzfhxFO4OP1nLXeLO2ktMIJMP9MlGv1Kog==
ARC-Authentication-Results: i=1; mx.microsoft.com 1; spf=pass (sender ip is
 216.228.112.36) smtp.rcpttodomain=redhat.com smtp.mailfrom=nvidia.com;
 dmarc=pass (p=none sp=none pct=100) action=none header.from=nvidia.com;
 dkim=none (message not signed); arc=none
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=Nvidia.com;
 s=selector2;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=KcnTnGk4MKucZ5I2jfckFtN2j5CTu6Fp125uCcm022A=;
 b=X21r1NwuSPLcPFFOdVf2KYbqz2TYx6VhHC0cJyUGkYmuj+WF841VXMx1HjhlkFtj5m4snw9C1TNEgmyJBj/g6CbZP6Ii7Nu2tJu7QkXfp5kt/4l+xwTT1p79eT6MSnppfol2MQL042kVmHsVidlcs3/t1kea78RPym6nQtbZlRwZ1SvYU+6bDrvBFOmHQY2iu7x5NyK22X55SHDfMY0AF0bEWssFh2j670QX1kCFjjuOM5vTPD7QxROl23yc21e/cYUER+PMyVKHoDw3dZNoEKx1CzlvKgHnBXTzuQEwY+I0EL+2c4JzSZUqRvJzDE8LTvjUFRxj+00mIO2vZTr6+Q==
Received: from BN6PR18CA0013.namprd18.prod.outlook.com (2603:10b6:404:121::23)
 by MWHPR1201MB0255.namprd12.prod.outlook.com (2603:10b6:301:4f::19) with
 Microsoft SMTP Server (version=TLS1_2,
 cipher=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384) id 15.20.4129.26; Tue, 18 May
 2021 19:21:37 +0000
Received: from BN8NAM11FT049.eop-nam11.prod.protection.outlook.com
 (2603:10b6:404:121:cafe::2a) by BN6PR18CA0013.outlook.office365.com
 (2603:10b6:404:121::23) with Microsoft SMTP Server (version=TLS1_2,
 cipher=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384) id 15.20.4129.25 via Frontend
 Transport; Tue, 18 May 2021 19:21:37 +0000
X-MS-Exchange-Authentication-Results: spf=pass (sender IP is 216.228.112.36)
 smtp.mailfrom=nvidia.com; redhat.com; dkim=none (message not signed)
 header.d=none;redhat.com; dmarc=pass action=none header.from=nvidia.com;
Received-SPF: Pass (protection.outlook.com: domain of nvidia.com designates
 216.228.112.36 as permitted sender) receiver=protection.outlook.com;
 client-ip=216.228.112.36; helo=mail.nvidia.com;
Received: from mail.nvidia.com (216.228.112.36) by
 BN8NAM11FT049.mail.protection.outlook.com (10.13.177.157) with Microsoft SMTP
 Server (version=TLS1_2, cipher=TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384) id
 15.20.4129.25 via Frontend Transport; Tue, 18 May 2021 19:21:37 +0000
Received: from HQMAIL101.nvidia.com (172.20.187.10) by HQMAIL101.nvidia.com
 (172.20.187.10) with Microsoft SMTP Server (TLS) id 15.0.1497.2; Tue, 18 May
 2021 19:21:36 +0000
Received: from r-arch-stor02.mtr.labs.mlnx (172.20.145.6) by mail.nvidia.com
 (172.20.187.10) with Microsoft SMTP Server id 15.0.1497.2 via Frontend
 Transport; Tue, 18 May 2021 19:21:34 +0000
From: Max Gurtovoy <mgurtovoy@nvidia.com>
To: <jgg@nvidia.com>, <cohuck@redhat.com>, <kvm@vger.kernel.org>,
        <alex.williamson@redhat.com>
CC: <oren@nvidia.com>, <eric.auger@redhat.com>,
        Max Gurtovoy <mgurtovoy@nvidia.com>
Subject: [PATCH 1/3] vfio/platform: fix module_put call in error flow
Date: Tue, 18 May 2021 22:21:31 +0300
Message-ID: <20210518192133.59195-1-mgurtovoy@nvidia.com>
X-Mailer: git-send-email 2.18.1
MIME-Version: 1.0
X-EOPAttributedMessage: 0
X-MS-PublicTrafficType: Email
X-MS-Office365-Filtering-Correlation-Id: 1bc92793-59fa-41a6-e77e-08d91a322806
X-MS-TrafficTypeDiagnostic: MWHPR1201MB0255:
X-Microsoft-Antispam-PRVS: 
 <MWHPR1201MB02552E1D75670B5263B1B322DE2C9@MWHPR1201MB0255.namprd12.prod.outlook.com>
X-MS-Oob-TLC-OOBClassifiers: OLM:5236;
X-MS-Exchange-SenderADCheck: 1
X-Microsoft-Antispam: BCL:0;
X-Microsoft-Antispam-Message-Info: 
 EnCdjKX8zbDKGqgC70wKUCfx9tsi/qrDVCqXETqGZFweAL35MjU2ZWTTnQolM4UCb3kGVcQ487Ntx2TaTVyqg+l1/yov3yWziTlDvgmX8Yc8/VwwwwJUURQqkERm/0loo0Uqu6ye9IfFzkQOk7ngxA+sJpGMdawGUfkFskxHCRhjxo2l02N9ZKXeYhneyvnYOs5o0tbshLoC7X/DecVAvPmbkypdReY9NUOVxFGA6rDN4LXwbCCsMQuVJRkyrguox3FpfIwBgkTh6Kz1inOSIJJouAtydmvckk9EO79ACyfI4wc3PgQPlllB6NwR/RWueVm62/saFdYEmEKg/JKrjTW27Nb4G3+0BYCO6F4pteLE4w0GX8zv0mRkFHd+0ANPc2AhvBVgfSiVYse3F36WKagEC47E+sN6vwOguWcLXrFfm7H7GexseMNQXJfo5BJURJ0lBrzT0hQLQ2Q6gH0nBxpWtRBVxmGfjRizoQ4o1f5vOfjaVxCwEXVIGAsDsRBQoNAlcaMyLngcHGzAa3poadQL4NOXOU4e5hqLnZMsk2SivyZvE4gTBB5J678FjnOs4yGaWOMHxMm+QY7INxT39srxhUN08DinVoA4nOoE4sizsuGX9u2px/FO8npiHnbJU1qp180EqF5Mm4/GgpmT9w==
X-Forefront-Antispam-Report: 
 CIP:216.228.112.36;CTRY:US;LANG:en;SCL:1;SRV:;IPV:NLI;SFV:NSPM;H:mail.nvidia.com;PTR:schybrid05.nvidia.com;CAT:NONE;SFS:(4636009)(376002)(396003)(136003)(39860400002)(346002)(36840700001)(46966006)(4326008)(356005)(70586007)(6666004)(7636003)(82740400003)(70206006)(36860700001)(426003)(5660300002)(478600001)(186003)(8676002)(4744005)(336012)(2616005)(47076005)(82310400003)(83380400001)(8936002)(54906003)(36756003)(316002)(36906005)(26005)(2906002)(86362001)(110136005)(107886003)(1076003);DIR:OUT;SFP:1101;
X-OriginatorOrg: Nvidia.com
X-MS-Exchange-CrossTenant-OriginalArrivalTime: 18 May 2021 19:21:37.0104
 (UTC)
X-MS-Exchange-CrossTenant-Network-Message-Id: 
 1bc92793-59fa-41a6-e77e-08d91a322806
X-MS-Exchange-CrossTenant-Id: 43083d15-7273-40c1-b7db-39efd9ccc17a
X-MS-Exchange-CrossTenant-OriginalAttributedTenantConnectingIp: 
 TenantId=43083d15-7273-40c1-b7db-39efd9ccc17a;Ip=[216.228.112.36];Helo=[mail.nvidia.com]
X-MS-Exchange-CrossTenant-AuthSource: 
 BN8NAM11FT049.eop-nam11.prod.protection.outlook.com
X-MS-Exchange-CrossTenant-AuthAs: Anonymous
X-MS-Exchange-CrossTenant-FromEntityHeader: HybridOnPrem
X-MS-Exchange-Transport-CrossTenantHeadersStamped: MWHPR1201MB0255
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

The ->parent_module is the one that use in try_module_get. It should
also be the one the we use in module_put during vfio_platform_open().

Fixes: 32a2d71c4e808 ("vfio: platform: introduce vfio-platform-base module")

Signed-off-by: Max Gurtovoy <mgurtovoy@nvidia.com>
---
 drivers/vfio/platform/vfio_platform_common.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/vfio/platform/vfio_platform_common.c b/drivers/vfio/platform/vfio_platform_common.c
index 361e5b57e369..470fcf7dac56 100644
--- a/drivers/vfio/platform/vfio_platform_common.c
+++ b/drivers/vfio/platform/vfio_platform_common.c
@@ -291,7 +291,7 @@ static int vfio_platform_open(struct vfio_device *core_vdev)
 	vfio_platform_regions_cleanup(vdev);
 err_reg:
 	mutex_unlock(&driver_lock);
-	module_put(THIS_MODULE);
+	module_put(vdev->parent_module);
 	return ret;
 }
 

From patchwork Tue May 18 19:21:32 2021
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Max Gurtovoy <mgurtovoy@nvidia.com>
X-Patchwork-Id: 12265523
Return-Path: <kvm-owner@kernel.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-19.1 required=3.0 tests=BAYES_00,DKIMWL_WL_HIGH,
	DKIM_SIGNED,DKIM_VALID,DKIM_VALID_AU,HEADER_FROM_DIFFERENT_DOMAINS,
	INCLUDES_CR_TRAILER,INCLUDES_PATCH,MAILING_LIST_MULTI,SPF_HELO_NONE,SPF_PASS,
	USER_AGENT_GIT autolearn=ham autolearn_force=no version=3.4.0
Received: from mail.kernel.org (mail.kernel.org [198.145.29.99])
	by smtp.lore.kernel.org (Postfix) with ESMTP id 30443C433B4
	for <kvm@archiver.kernel.org>; Tue, 18 May 2021 19:21:42 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [23.128.96.18])
	by mail.kernel.org (Postfix) with ESMTP id 07F5160FE8
	for <kvm@archiver.kernel.org>; Tue, 18 May 2021 19:21:42 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1351776AbhERTW7 (ORCPT <rfc822;kvm@archiver.kernel.org>);
        Tue, 18 May 2021 15:22:59 -0400
Received: from mail-dm6nam10on2086.outbound.protection.outlook.com
 ([40.107.93.86]:37533
        "EHLO NAM10-DM6-obe.outbound.protection.outlook.com"
        rhost-flags-OK-OK-OK-FAIL) by vger.kernel.org with ESMTP
        id S237739AbhERTW6 (ORCPT <rfc822;kvm@vger.kernel.org>);
        Tue, 18 May 2021 15:22:58 -0400
ARC-Seal: i=1; a=rsa-sha256; s=arcselector9901; d=microsoft.com; cv=none;
 b=kxgNjnliBfxYOisxHl6TTdXBVieY0t3Fm9fECVfdGjefEHnetgjAwD+c1u+pp6vwuHONF4WJIDWNfct9rnjsrY6YfPwpBFfIu973kAYzlcT3rrFzG66BSejEEsgUIpRu+y/TMl/MWERJTctjBQSxU2Y8yZaRfSd4/40xox+8+pHUqsHycSTznawuvVRWCg3RMfAhBMQBReS0Fo/aVRUIEZuTS+RrGHpqC2YdxzAKJ+ZvnOUzqpsEY449jwHkJ41+mR+e7nxETsF+Fv4chlksENZkJWVzht7xcLrEEJDqN3AToPEwZQjQ40Eb1V0klG9nzRgoGc41md78kc65+uQ9CA==
ARC-Message-Signature: i=1; a=rsa-sha256; c=relaxed/relaxed; d=microsoft.com;
 s=arcselector9901;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=uPvc4jkBAktBBlo93h8/ngO641pYp8KyDgeGu9dY7/0=;
 b=XSHjXU2skOENz3RWqVa+FDk1I+DZBDwrC2OI8849kuBwGkiGiRa2f895DtknsnEnZqofOV/CrFj9P/I+RStxeLJZuiDiNHOMBSgLZs5xoDOrLgCQCKAIyrTe+Q8+uDvWSHW9v24kFMVsBeFQZTOxVlQLTlFOGmPpM9ds4vAIGbrRMjmrWYfoHZ5JKJXM2N4O0YvkIQEWoYhoPVKtSaUK/uW79rJAtv07H1MrqeUKGzf9Os+3outb6O7FEpGzGUX/LfSK4hFZwfPnvM9r3jWK4TROFKnNGKAsJ9dOJRKh0mhSNnZaTfYY5rvSmz6DYgg8xri5IjQQJnuddh0d3heiBQ==
ARC-Authentication-Results: i=1; mx.microsoft.com 1; spf=pass (sender ip is
 216.228.112.35) smtp.rcpttodomain=redhat.com smtp.mailfrom=nvidia.com;
 dmarc=pass (p=none sp=none pct=100) action=none header.from=nvidia.com;
 dkim=none (message not signed); arc=none
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=Nvidia.com;
 s=selector2;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=uPvc4jkBAktBBlo93h8/ngO641pYp8KyDgeGu9dY7/0=;
 b=XeF6k88oy3IH1MzIiUH+aCHG18R05Rpg9PmNy0oyufBkcwSkXGh8f4/KPXTcOKYUcr1TA0sszvNDoY1MkBbjs8kxG37JYQxLAJyClyDLWWpOYHh2RygK9dVoPR5z/j3QPvNgYapQ1G4AQVhhbZ9bvs5aMTIo/fm/pH4/T/rfaYM7jTzDbkjhpq/MTP23oXDAQW0tSrJin6t4PvPYlHloHno6kYy91mixbaIJzIkbVi67FpPprfkolOxtddPL5mHsYY3hKnonPtxmCGoZ5Y2rCon/JRX0GvQByy8pQGq5iZ0dRYm8NTwJnW063kzz3Eao+oOtomyAh3X/tdyhJkvkVQ==
Received: from BN1PR12CA0014.namprd12.prod.outlook.com (2603:10b6:408:e1::19)
 by MN2PR12MB3614.namprd12.prod.outlook.com (2603:10b6:208:c6::17) with
 Microsoft SMTP Server (version=TLS1_2,
 cipher=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384) id 15.20.4129.28; Tue, 18 May
 2021 19:21:39 +0000
Received: from BN8NAM11FT040.eop-nam11.prod.protection.outlook.com
 (2603:10b6:408:e1:cafe::ed) by BN1PR12CA0014.outlook.office365.com
 (2603:10b6:408:e1::19) with Microsoft SMTP Server (version=TLS1_2,
 cipher=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384) id 15.20.4108.25 via Frontend
 Transport; Tue, 18 May 2021 19:21:39 +0000
X-MS-Exchange-Authentication-Results: spf=pass (sender IP is 216.228.112.35)
 smtp.mailfrom=nvidia.com; redhat.com; dkim=none (message not signed)
 header.d=none;redhat.com; dmarc=pass action=none header.from=nvidia.com;
Received-SPF: Pass (protection.outlook.com: domain of nvidia.com designates
 216.228.112.35 as permitted sender) receiver=protection.outlook.com;
 client-ip=216.228.112.35; helo=mail.nvidia.com;
Received: from mail.nvidia.com (216.228.112.35) by
 BN8NAM11FT040.mail.protection.outlook.com (10.13.177.166) with Microsoft SMTP
 Server (version=TLS1_2, cipher=TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384) id
 15.20.4129.25 via Frontend Transport; Tue, 18 May 2021 19:21:39 +0000
Received: from HQMAIL105.nvidia.com (172.20.187.12) by HQMAIL111.nvidia.com
 (172.20.187.18) with Microsoft SMTP Server (TLS) id 15.0.1497.2; Tue, 18 May
 2021 19:21:38 +0000
Received: from HQMAIL101.nvidia.com (172.20.187.10) by HQMAIL105.nvidia.com
 (172.20.187.12) with Microsoft SMTP Server (TLS) id 15.0.1497.2; Tue, 18 May
 2021 19:21:38 +0000
Received: from r-arch-stor02.mtr.labs.mlnx (172.20.145.6) by mail.nvidia.com
 (172.20.187.10) with Microsoft SMTP Server id 15.0.1497.2 via Frontend
 Transport; Tue, 18 May 2021 19:21:36 +0000
From: Max Gurtovoy <mgurtovoy@nvidia.com>
To: <jgg@nvidia.com>, <cohuck@redhat.com>, <kvm@vger.kernel.org>,
        <alex.williamson@redhat.com>
CC: <oren@nvidia.com>, <eric.auger@redhat.com>,
        Max Gurtovoy <mgurtovoy@nvidia.com>
Subject: [PATCH 2/3] vfio: centralize module refcount in subsystem layer
Date: Tue, 18 May 2021 22:21:32 +0300
Message-ID: <20210518192133.59195-2-mgurtovoy@nvidia.com>
X-Mailer: git-send-email 2.18.1
In-Reply-To: <20210518192133.59195-1-mgurtovoy@nvidia.com>
References: <20210518192133.59195-1-mgurtovoy@nvidia.com>
MIME-Version: 1.0
X-EOPAttributedMessage: 0
X-MS-PublicTrafficType: Email
X-MS-Office365-Filtering-Correlation-Id: 94d8c1f5-27b5-4a03-3427-08d91a322952
X-MS-TrafficTypeDiagnostic: MN2PR12MB3614:
X-Microsoft-Antispam-PRVS: 
 <MN2PR12MB3614F48C90EC40387F468969DE2C9@MN2PR12MB3614.namprd12.prod.outlook.com>
X-MS-Oob-TLC-OOBClassifiers: OLM:792;
X-MS-Exchange-SenderADCheck: 1
X-Microsoft-Antispam: BCL:0;
X-Microsoft-Antispam-Message-Info: 
 kAO+HgQECi/+b1qryczQ1ehz9rHjpji3y8YvCWY6twLs19xkaAhU3+9vsODRzzF1k3thqdcvFmiWuL6pmf3i0HUpVsEJyccadDbCMSn7dSOMP9Dect+/Gab2GZyVvyj4pfZm3ACBoVK/PqDyx0IqS6Xhpt9NsAyz2Y7cY5a6UCIKFOAXtRVJq03hSPX3Ioq/2Ij5dI6dPT8L0KN/tNjVEwyWX9YkCVg6uiem/YSc0f5nplzvNRjYd1RTVzdVeJkiS2HXGdnL6Fiy7hcwUgwly5dQ1wnGlKn65qxoGHYo4E74jU+65zEQA4mrlbuSxx+oqWQGLe0AdgSki7BaO9NGfdDr8c8TDt2rLKrplI7T8mrcHANHlXfBXy+1RxynRlQTdHtOD4jgsc4q547rgfeTbftoRcp3Ypxav2Mv1Z8lTsaCGCKH3iu4/ptta9fvvEDWocOe+2ai2gVSvncXjME0coTk7q6udXdYLHm6UodyWC5nsWcSsnUWSZUw1BmjwLTbAQRm5bY2RBddKwWqoE9qUxczKhexh3IVRHNODAfDQjW5FqzdSTF8uB2hYWGlqphbEwvYtuPmRy9o4uKfEA8p23Xi56YlaUXjcQplM7sbbncz+w3+zF5WtpfDrkpH3xTNTJRQhzONspUHIFRlIGqfb0G0AA2bd1BhAj6L1CESMr8=
X-Forefront-Antispam-Report: 
 CIP:216.228.112.35;CTRY:US;LANG:en;SCL:1;SRV:;IPV:NLI;SFV:NSPM;H:mail.nvidia.com;PTR:schybrid04.nvidia.com;CAT:NONE;SFS:(4636009)(39860400002)(136003)(376002)(396003)(346002)(36840700001)(46966006)(6666004)(4326008)(1076003)(83380400001)(2906002)(36756003)(82740400003)(86362001)(426003)(2616005)(47076005)(186003)(336012)(26005)(36906005)(316002)(110136005)(54906003)(356005)(107886003)(8676002)(70586007)(7636003)(70206006)(8936002)(478600001)(82310400003)(5660300002)(36860700001);DIR:OUT;SFP:1101;
X-OriginatorOrg: Nvidia.com
X-MS-Exchange-CrossTenant-OriginalArrivalTime: 18 May 2021 19:21:39.1876
 (UTC)
X-MS-Exchange-CrossTenant-Network-Message-Id: 
 94d8c1f5-27b5-4a03-3427-08d91a322952
X-MS-Exchange-CrossTenant-Id: 43083d15-7273-40c1-b7db-39efd9ccc17a
X-MS-Exchange-CrossTenant-OriginalAttributedTenantConnectingIp: 
 TenantId=43083d15-7273-40c1-b7db-39efd9ccc17a;Ip=[216.228.112.35];Helo=[mail.nvidia.com]
X-MS-Exchange-CrossTenant-AuthSource: 
 BN8NAM11FT040.eop-nam11.prod.protection.outlook.com
X-MS-Exchange-CrossTenant-AuthAs: Anonymous
X-MS-Exchange-CrossTenant-FromEntityHeader: HybridOnPrem
X-MS-Exchange-Transport-CrossTenantHeadersStamped: MN2PR12MB3614
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

Remove code duplication and move module refcounting to the subsystem
module.

Signed-off-by: Max Gurtovoy <mgurtovoy@nvidia.com>
---
 drivers/vfio/fsl-mc/vfio_fsl_mc.c            | 16 +++-------------
 drivers/vfio/mdev/vfio_mdev.c                | 13 +------------
 drivers/vfio/pci/vfio_pci.c                  |  7 -------
 drivers/vfio/platform/vfio_platform_common.c |  6 ------
 drivers/vfio/vfio.c                          | 10 ++++++++++
 5 files changed, 14 insertions(+), 38 deletions(-)

diff --git a/drivers/vfio/fsl-mc/vfio_fsl_mc.c b/drivers/vfio/fsl-mc/vfio_fsl_mc.c
index 980e59551301..90cad109583b 100644
--- a/drivers/vfio/fsl-mc/vfio_fsl_mc.c
+++ b/drivers/vfio/fsl-mc/vfio_fsl_mc.c
@@ -140,26 +140,18 @@ static int vfio_fsl_mc_open(struct vfio_device *core_vdev)
 {
 	struct vfio_fsl_mc_device *vdev =
 		container_of(core_vdev, struct vfio_fsl_mc_device, vdev);
-	int ret;
-
-	if (!try_module_get(THIS_MODULE))
-		return -ENODEV;
+	int ret = 0;
 
 	mutex_lock(&vdev->reflck->lock);
 	if (!vdev->refcnt) {
 		ret = vfio_fsl_mc_regions_init(vdev);
 		if (ret)
-			goto err_reg_init;
+			goto out;
 	}
 	vdev->refcnt++;
-
+out:
 	mutex_unlock(&vdev->reflck->lock);
 
-	return 0;
-
-err_reg_init:
-	mutex_unlock(&vdev->reflck->lock);
-	module_put(THIS_MODULE);
 	return ret;
 }
 
@@ -196,8 +188,6 @@ static void vfio_fsl_mc_release(struct vfio_device *core_vdev)
 	}
 
 	mutex_unlock(&vdev->reflck->lock);
-
-	module_put(THIS_MODULE);
 }
 
 static long vfio_fsl_mc_ioctl(struct vfio_device *core_vdev,
diff --git a/drivers/vfio/mdev/vfio_mdev.c b/drivers/vfio/mdev/vfio_mdev.c
index 922729071c5a..5ef4815609ed 100644
--- a/drivers/vfio/mdev/vfio_mdev.c
+++ b/drivers/vfio/mdev/vfio_mdev.c
@@ -26,19 +26,10 @@ static int vfio_mdev_open(struct vfio_device *core_vdev)
 	struct mdev_device *mdev = to_mdev_device(core_vdev->dev);
 	struct mdev_parent *parent = mdev->type->parent;
 
-	int ret;
-
 	if (unlikely(!parent->ops->open))
 		return -EINVAL;
 
-	if (!try_module_get(THIS_MODULE))
-		return -ENODEV;
-
-	ret = parent->ops->open(mdev);
-	if (ret)
-		module_put(THIS_MODULE);
-
-	return ret;
+	return parent->ops->open(mdev);
 }
 
 static void vfio_mdev_release(struct vfio_device *core_vdev)
@@ -48,8 +39,6 @@ static void vfio_mdev_release(struct vfio_device *core_vdev)
 
 	if (likely(parent->ops->release))
 		parent->ops->release(mdev);
-
-	module_put(THIS_MODULE);
 }
 
 static long vfio_mdev_unlocked_ioctl(struct vfio_device *core_vdev,
diff --git a/drivers/vfio/pci/vfio_pci.c b/drivers/vfio/pci/vfio_pci.c
index bd7c482c948a..f6729baa1bf4 100644
--- a/drivers/vfio/pci/vfio_pci.c
+++ b/drivers/vfio/pci/vfio_pci.c
@@ -558,8 +558,6 @@ static void vfio_pci_release(struct vfio_device *core_vdev)
 	}
 
 	mutex_unlock(&vdev->reflck->lock);
-
-	module_put(THIS_MODULE);
 }
 
 static int vfio_pci_open(struct vfio_device *core_vdev)
@@ -568,9 +566,6 @@ static int vfio_pci_open(struct vfio_device *core_vdev)
 		container_of(core_vdev, struct vfio_pci_device, vdev);
 	int ret = 0;
 
-	if (!try_module_get(THIS_MODULE))
-		return -ENODEV;
-
 	mutex_lock(&vdev->reflck->lock);
 
 	if (!vdev->refcnt) {
@@ -584,8 +579,6 @@ static int vfio_pci_open(struct vfio_device *core_vdev)
 	vdev->refcnt++;
 error:
 	mutex_unlock(&vdev->reflck->lock);
-	if (ret)
-		module_put(THIS_MODULE);
 	return ret;
 }
 
diff --git a/drivers/vfio/platform/vfio_platform_common.c b/drivers/vfio/platform/vfio_platform_common.c
index 470fcf7dac56..703164df7637 100644
--- a/drivers/vfio/platform/vfio_platform_common.c
+++ b/drivers/vfio/platform/vfio_platform_common.c
@@ -241,8 +241,6 @@ static void vfio_platform_release(struct vfio_device *core_vdev)
 	}
 
 	mutex_unlock(&driver_lock);
-
-	module_put(vdev->parent_module);
 }
 
 static int vfio_platform_open(struct vfio_device *core_vdev)
@@ -251,9 +249,6 @@ static int vfio_platform_open(struct vfio_device *core_vdev)
 		container_of(core_vdev, struct vfio_platform_device, vdev);
 	int ret;
 
-	if (!try_module_get(vdev->parent_module))
-		return -ENODEV;
-
 	mutex_lock(&driver_lock);
 
 	if (!vdev->refcnt) {
@@ -291,7 +286,6 @@ static int vfio_platform_open(struct vfio_device *core_vdev)
 	vfio_platform_regions_cleanup(vdev);
 err_reg:
 	mutex_unlock(&driver_lock);
-	module_put(vdev->parent_module);
 	return ret;
 }
 
diff --git a/drivers/vfio/vfio.c b/drivers/vfio/vfio.c
index 5e631c359ef2..02cc51ce6891 100644
--- a/drivers/vfio/vfio.c
+++ b/drivers/vfio/vfio.c
@@ -1369,8 +1369,14 @@ static int vfio_group_get_device_fd(struct vfio_group *group, char *buf)
 	if (IS_ERR(device))
 		return PTR_ERR(device);
 
+	if (!try_module_get(device->dev->driver->owner)) {
+		vfio_device_put(device);
+		return -ENODEV;
+	}
+
 	ret = device->ops->open(device);
 	if (ret) {
+		module_put(device->dev->driver->owner);
 		vfio_device_put(device);
 		return ret;
 	}
@@ -1382,6 +1388,7 @@ static int vfio_group_get_device_fd(struct vfio_group *group, char *buf)
 	ret = get_unused_fd_flags(O_CLOEXEC);
 	if (ret < 0) {
 		device->ops->release(device);
+		module_put(device->dev->driver->owner);
 		vfio_device_put(device);
 		return ret;
 	}
@@ -1392,6 +1399,7 @@ static int vfio_group_get_device_fd(struct vfio_group *group, char *buf)
 		put_unused_fd(ret);
 		ret = PTR_ERR(filep);
 		device->ops->release(device);
+		module_put(device->dev->driver->owner);
 		vfio_device_put(device);
 		return ret;
 	}
@@ -1550,6 +1558,8 @@ static int vfio_device_fops_release(struct inode *inode, struct file *filep)
 
 	device->ops->release(device);
 
+	module_put(device->dev->driver->owner);
+
 	vfio_group_try_dissolve_container(device->group);
 
 	vfio_device_put(device);

From patchwork Tue May 18 19:21:33 2021
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Max Gurtovoy <mgurtovoy@nvidia.com>
X-Patchwork-Id: 12265525
Return-Path: <kvm-owner@kernel.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-19.1 required=3.0 tests=BAYES_00,DKIMWL_WL_HIGH,
	DKIM_SIGNED,DKIM_VALID,DKIM_VALID_AU,HEADER_FROM_DIFFERENT_DOMAINS,
	INCLUDES_CR_TRAILER,INCLUDES_PATCH,MAILING_LIST_MULTI,SPF_HELO_NONE,SPF_PASS,
	USER_AGENT_GIT autolearn=ham autolearn_force=no version=3.4.0
Received: from mail.kernel.org (mail.kernel.org [198.145.29.99])
	by smtp.lore.kernel.org (Postfix) with ESMTP id 32F2FC433B4
	for <kvm@archiver.kernel.org>; Tue, 18 May 2021 19:21:46 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [23.128.96.18])
	by mail.kernel.org (Postfix) with ESMTP id 0AB7860FE8
	for <kvm@archiver.kernel.org>; Tue, 18 May 2021 19:21:46 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1351792AbhERTXD (ORCPT <rfc822;kvm@archiver.kernel.org>);
        Tue, 18 May 2021 15:23:03 -0400
Received: from mail-dm6nam10on2083.outbound.protection.outlook.com
 ([40.107.93.83]:61793
        "EHLO NAM10-DM6-obe.outbound.protection.outlook.com"
        rhost-flags-OK-OK-OK-FAIL) by vger.kernel.org with ESMTP
        id S237739AbhERTXC (ORCPT <rfc822;kvm@vger.kernel.org>);
        Tue, 18 May 2021 15:23:02 -0400
ARC-Seal: i=1; a=rsa-sha256; s=arcselector9901; d=microsoft.com; cv=none;
 b=KFaSBuxe3b/2aFfVlk/DzCGwXP8NUaBavSJoTsW2CovLofFkaBW4s31/i92HuuCDF6HAFwzvz342+c82zCvN6Dx0MtP99dbRMdb1Vp/W5CCENa/Fx4657OmOFm665k4NyyFznCQSZBJ2WSF7IEJmvOK8Rlq/to4fo8d3ZCfVyL76t+7/t0cfX37rx6alJXLy2Q/tODonDeVlH1mgvjpaeuMj3vfhBQO+q9Lt1IhLWEfel+5uQhE3RB4zB8rbiLXIe2f6slm9s6cFn+SMj5ZBLWroDOIhxlgwculPnOafZJ9v4KHCqsvRS6n1mG1FxjDXWj3MdgEsd3LUVjB2eSLVKw==
ARC-Message-Signature: i=1; a=rsa-sha256; c=relaxed/relaxed; d=microsoft.com;
 s=arcselector9901;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=wBwLHiw9ajzyRzm1MlIY1rVbZsPqz/BnhHZL9bDzJEM=;
 b=R75H4IjRFcvZc7pFaMY/9Et81dJqh8gGqkhjtyVyinJYZFbQSZBRi4JY/4UYJBEtd0aBJadB9R65ed7b4OPIJ+OZ1BONrhGngqNFZFGs+b6S5lmcbTQmIe7d/EYzNUfsrOlXIfMv1fOwbUUChqZPMS8PtPK0O8TmaIdvqsgJs6W+HKmGR4ljxdtsXUsoVlwzXYJImoauoEaxZYrE4Rh8P5dddQeBkmVDi53++YdhQhyVG3UZHeSnhKZfzaHeGjmL0HNZvGnlf3E9OPOnVqqe7zNSe9KclqyBeHy0i+nxrr5ZxAIouUOuNKffeT3MZCuYX9sq7Oy8QRnDMbaiiNWwIg==
ARC-Authentication-Results: i=1; mx.microsoft.com 1; spf=pass (sender ip is
 216.228.112.32) smtp.rcpttodomain=redhat.com smtp.mailfrom=nvidia.com;
 dmarc=pass (p=none sp=none pct=100) action=none header.from=nvidia.com;
 dkim=none (message not signed); arc=none
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=Nvidia.com;
 s=selector2;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=wBwLHiw9ajzyRzm1MlIY1rVbZsPqz/BnhHZL9bDzJEM=;
 b=jOigKUJfqK5cKUOMNOP/L55qMq3zG36gda0BQQLpLg/d11pwOKauBtN8wlIgksKUFwy7pg/+e9I4SAZ+woiBHlTPZ5HkhpsLZoJP8qRSSt2FvtjGKc1drObtg8V+mlKfSQlJl6cmDaT7eBvBtobptDPAzO37YHJrceYkusv0HYTpPJbwHejJdrv0HDhjtoHnaXySlWh0JW9z+y4f43+1cqTgb+jQNemfuUT/7RdZHabomsOsAualzGNnTfwaVBAiF7kKNU7RJThe/vMu/KLm2KiC/xjNPoy9NqbnEaJ8bJggVck6lyi6cIJp7W8OWIMoizAlSX8eJjr3gd3jvjlhng==
Received: from DM6PR06CA0038.namprd06.prod.outlook.com (2603:10b6:5:54::15) by
 BL0PR12MB4706.namprd12.prod.outlook.com (2603:10b6:208:82::25) with Microsoft
 SMTP Server (version=TLS1_2, cipher=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384) id
 15.20.4129.25; Tue, 18 May 2021 19:21:41 +0000
Received: from DM6NAM11FT048.eop-nam11.prod.protection.outlook.com
 (2603:10b6:5:54:cafe::ff) by DM6PR06CA0038.outlook.office365.com
 (2603:10b6:5:54::15) with Microsoft SMTP Server (version=TLS1_2,
 cipher=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384) id 15.20.4129.25 via Frontend
 Transport; Tue, 18 May 2021 19:21:41 +0000
X-MS-Exchange-Authentication-Results: spf=pass (sender IP is 216.228.112.32)
 smtp.mailfrom=nvidia.com; redhat.com; dkim=none (message not signed)
 header.d=none;redhat.com; dmarc=pass action=none header.from=nvidia.com;
Received-SPF: Pass (protection.outlook.com: domain of nvidia.com designates
 216.228.112.32 as permitted sender) receiver=protection.outlook.com;
 client-ip=216.228.112.32; helo=mail.nvidia.com;
Received: from mail.nvidia.com (216.228.112.32) by
 DM6NAM11FT048.mail.protection.outlook.com (10.13.173.114) with Microsoft SMTP
 Server (version=TLS1_2, cipher=TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384) id
 15.20.4129.25 via Frontend Transport; Tue, 18 May 2021 19:21:41 +0000
Received: from HQMAIL101.nvidia.com (172.20.187.10) by HQMAIL109.nvidia.com
 (172.20.187.15) with Microsoft SMTP Server (TLS) id 15.0.1497.2; Tue, 18 May
 2021 12:21:40 -0700
Received: from r-arch-stor02.mtr.labs.mlnx (172.20.145.6) by mail.nvidia.com
 (172.20.187.10) with Microsoft SMTP Server id 15.0.1497.2 via Frontend
 Transport; Tue, 18 May 2021 19:21:39 +0000
From: Max Gurtovoy <mgurtovoy@nvidia.com>
To: <jgg@nvidia.com>, <cohuck@redhat.com>, <kvm@vger.kernel.org>,
        <alex.williamson@redhat.com>
CC: <oren@nvidia.com>, <eric.auger@redhat.com>,
        Max Gurtovoy <mgurtovoy@nvidia.com>
Subject: [PATCH 3/3] vfio/platform: remove unneeded parent_module attribute
Date: Tue, 18 May 2021 22:21:33 +0300
Message-ID: <20210518192133.59195-3-mgurtovoy@nvidia.com>
X-Mailer: git-send-email 2.18.1
In-Reply-To: <20210518192133.59195-1-mgurtovoy@nvidia.com>
References: <20210518192133.59195-1-mgurtovoy@nvidia.com>
MIME-Version: 1.0
X-EOPAttributedMessage: 0
X-MS-PublicTrafficType: Email
X-MS-Office365-Filtering-Correlation-Id: 79ca27cc-1201-4c76-d29b-08d91a322a68
X-MS-TrafficTypeDiagnostic: BL0PR12MB4706:
X-Microsoft-Antispam-PRVS: 
 <BL0PR12MB4706A36923D66F02D68D8EEFDE2C9@BL0PR12MB4706.namprd12.prod.outlook.com>
X-MS-Oob-TLC-OOBClassifiers: OLM:1169;
X-MS-Exchange-SenderADCheck: 1
X-Microsoft-Antispam: BCL:0;
X-Microsoft-Antispam-Message-Info: 
 VuSZqT/KJS2yeWV2P4HtYl4CeUpgkAitXTMr3GaBRqhVSZq3aVwPVVUc1BIAdF0luQNV0AdInD/v0YNT8fL1CAdFU/0Kjf6ZS7VYVysJDYtbOtgZciY/1QS3htD+VGzNTZdF+4jw5gfWMm6TR5E+uXe+5lUJAAidKAqAU+x36komMMpvbzyw0qd2IswKQUj1WhdE4o/xgkjtwIQ4niZfNnnFn4jDXQ80HRocQ8S5hsDe+hv9M4FCwVEh3isU34xupzdQGTe72WTnXx57hlFDwdpbNgJVmEh1pJq4PMCsagyjWyJg6R1cfR0OqnhIZnA+XoTYm4ChvRIQh1247G3mjiAK6VKfcVeh2Iir8vLrdmGAdwTd59RqPkf0AFanSurrxfpqEzTUxIJ+zhsYznTtI3Gux8H/ucrrjI7bLo3z6gsn48/G6yA5DA2PsGSWBPfc8Xjmp/hK0absZMSUUPIJPA97ZihNlphwTl5vY3POKqZpdXxha0dv64/edolMlzR8qvvxbJdhr6OvbkXxs16Rsabv4LiQrSBglqNPEUWBeD0gULbmWb7kdefSAlYmGhe6keD2XIvbS52l2ax+i15ZHb+oVBznNftFvYxCQoR8qkXhytU5i6f/q5XKvsKpTNja3/WybMPOI3M6tNr10Gwa5+XHEgRDUjIJaSvOc1zHWw4=
X-Forefront-Antispam-Report: 
 CIP:216.228.112.32;CTRY:US;LANG:en;SCL:1;SRV:;IPV:NLI;SFV:NSPM;H:mail.nvidia.com;PTR:schybrid01.nvidia.com;CAT:NONE;SFS:(4636009)(396003)(346002)(376002)(39860400002)(136003)(36840700001)(46966006)(110136005)(186003)(426003)(336012)(36756003)(1076003)(7636003)(316002)(83380400001)(356005)(82310400003)(82740400003)(47076005)(54906003)(107886003)(2906002)(6666004)(70586007)(36860700001)(86362001)(4326008)(70206006)(8676002)(8936002)(26005)(478600001)(5660300002)(2616005);DIR:OUT;SFP:1101;
X-OriginatorOrg: Nvidia.com
X-MS-Exchange-CrossTenant-OriginalArrivalTime: 18 May 2021 19:21:41.0856
 (UTC)
X-MS-Exchange-CrossTenant-Network-Message-Id: 
 79ca27cc-1201-4c76-d29b-08d91a322a68
X-MS-Exchange-CrossTenant-Id: 43083d15-7273-40c1-b7db-39efd9ccc17a
X-MS-Exchange-CrossTenant-OriginalAttributedTenantConnectingIp: 
 TenantId=43083d15-7273-40c1-b7db-39efd9ccc17a;Ip=[216.228.112.32];Helo=[mail.nvidia.com]
X-MS-Exchange-CrossTenant-AuthSource: 
 DM6NAM11FT048.eop-nam11.prod.protection.outlook.com
X-MS-Exchange-CrossTenant-AuthAs: Anonymous
X-MS-Exchange-CrossTenant-FromEntityHeader: HybridOnPrem
X-MS-Exchange-Transport-CrossTenantHeadersStamped: BL0PR12MB4706
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

The vfio core driver is now taking refcount on the provider drivers,
remove redundant parent_module attribute from vfio_platform_device
structure.

Signed-off-by: Max Gurtovoy <mgurtovoy@nvidia.com>
---
 drivers/vfio/platform/vfio_amba.c             | 1 -
 drivers/vfio/platform/vfio_platform.c         | 1 -
 drivers/vfio/platform/vfio_platform_private.h | 1 -
 3 files changed, 3 deletions(-)

diff --git a/drivers/vfio/platform/vfio_amba.c b/drivers/vfio/platform/vfio_amba.c
index f970eb2a999f..badfffea14fb 100644
--- a/drivers/vfio/platform/vfio_amba.c
+++ b/drivers/vfio/platform/vfio_amba.c
@@ -59,7 +59,6 @@ static int vfio_amba_probe(struct amba_device *adev, const struct amba_id *id)
 	vdev->flags = VFIO_DEVICE_FLAGS_AMBA;
 	vdev->get_resource = get_amba_resource;
 	vdev->get_irq = get_amba_irq;
-	vdev->parent_module = THIS_MODULE;
 	vdev->reset_required = false;
 
 	ret = vfio_platform_probe_common(vdev, &adev->dev);
diff --git a/drivers/vfio/platform/vfio_platform.c b/drivers/vfio/platform/vfio_platform.c
index e4027799a154..68a1c87066d7 100644
--- a/drivers/vfio/platform/vfio_platform.c
+++ b/drivers/vfio/platform/vfio_platform.c
@@ -50,7 +50,6 @@ static int vfio_platform_probe(struct platform_device *pdev)
 	vdev->flags = VFIO_DEVICE_FLAGS_PLATFORM;
 	vdev->get_resource = get_platform_resource;
 	vdev->get_irq = get_platform_irq;
-	vdev->parent_module = THIS_MODULE;
 	vdev->reset_required = reset_required;
 
 	ret = vfio_platform_probe_common(vdev, &pdev->dev);
diff --git a/drivers/vfio/platform/vfio_platform_private.h b/drivers/vfio/platform/vfio_platform_private.h
index a5ba82c8cbc3..dfb834c13659 100644
--- a/drivers/vfio/platform/vfio_platform_private.h
+++ b/drivers/vfio/platform/vfio_platform_private.h
@@ -50,7 +50,6 @@ struct vfio_platform_device {
 	u32				num_irqs;
 	int				refcnt;
 	struct mutex			igate;
-	struct module			*parent_module;
 	const char			*compat;
 	const char			*acpihid;
 	struct module			*reset_module;
